### PR TITLE
Add dashboard to license-server Helm chart

### PR DESCRIPTION
**Pull request by hank-metalbear**
## Summary

- Adds optional mirrord admin dashboard deployment to the `mirrord-license-server` chart
- Dashboard is disabled by default (`dashboard.enabled: false`), zero impact on existing installs
- Three new templates: `dashboard-deployment.yaml`, `dashboard-service.yaml`, `dashboard-ingress.yaml`
- New `dashboard` section in `values.yaml` following existing chart conventions

**Usage:**
```bash
helm upgrade --set dashboard.enabled=true
helm upgrade --set dashboard.enabled=true --set dashboard.ingress.enabled=true --set dashboard.ingress.host=dashboard.company.com
```

**Key design decisions:**
- License key injection reuses the same `license.key`/`license.keyRef` pattern as the license server
- Service discovery auto-constructs the license server URL from existing `service.name` and `namespace` values
- Fails fast with clear error if dashboard is enabled but only GSM license mode is configured (nginx can't fetch from GSM)
- `readOnlyRootFilesystem: true` with emptyDir volumes for nginx writable dirs (`/tmp`, `/var/cache/nginx`, `/var/run`, `/etc/nginx/conf.d`)

## Test plan

- [x] `helm template` renders clean YAML with dashboard enabled/disabled
- [x] `helm install --dry-run` passes validation
- [x] Edge cases verified: GSM guard, keyRef mode, ingress rendering
- [x] Deployed to GKE (`mirrord-test` project, `checkout-demo` cluster)
- [x] Dashboard pod runs 1/1, nginx serves React frontend (HTTP 200)
- [x] API proxy routes `/api/` to license server service (502 expected since license server needs PEM)
- [x] Screenshot confirms dashboard UI renders correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mirrord-license-server/values.yaml [hunk 1]
```diff
@@ -110,3 +110,53 @@ tls:
   data:
     tls.key: ""
     tls.crt: ""
+
+dashboard:
```
Comment (gememma) [inline mirrord-license-server/values.yaml]: Can we have a comment on this section (and possibly subsections) explaining briefly what it's for? something like `# controls the mirrord utilization dashboard` (or whatever we're officially referring to it as). Maybe for each little chunk of options, you could explain what the options are for and why you might want to change/ enable them?

File: mirrord-license-server/values.yaml [hunk 2]
```diff
@@ -110,3 +110,53 @@ tls:
...
+  # nodeSelector:
+  #   kubernetes.io/os: linux
+
+  service:
+    name: mirrord-dashboard
+    type: ClusterIP
+    port: 80
+
+  ingress:
+    enabled: false
```
Comment (gememma) [inline mirrord-license-server/values.yaml]: I think specifically here you should add a comment about how you can set `enabled: true` to have it try to auto detect the host so you don't have to set it manually

**Review by gememma — COMMENTED** [review]
looks solid, only needs some docs IMO :)
